### PR TITLE
Update of concept "numbers" about.md & links.json

### PR DESCRIPTION
**Pull request by cydeb33**
Hello, I have updated the content of the about.md file according to the issue opened here https://github.com/exercism/javascript/issues/1505. I hope my work is relevant.
I'll update anything necessary based on your review. 
I noticed a weird behavior but I don't know if it's a bug, on line 60 for example, when creating a link "[`infinity`][infinity]", it didn't show as a link on my fork, but wrote it as is.
Thank you

### CODE REVIEW

File: concepts/numbers/about.md [hunk 1]
```diff
@@ -55,6 +55,73 @@ There are three types of maximum (and minimum / maximum negative) values for num
...
+Here are some examples on what you can and can not do :
+```javascript
+int x4 = 5_______2;        // OK (decimal literal)
+int x2 = 5_2;              // OK (decimal literal)
+float pi1 = 3_.1415F;      // Invalid; cannot put underscores adjacent to a decimal point
+int x11 = 052_;            // Invalid; cannot put underscores at the end of a number
+int x5 = 0_x52;            // Invalid; cannot put underscores in the 0x radix prefix
+int x6 = 0x_52;            // Invalid; cannot put underscores at the beginning of a number
+long socialSecurityNumber1 = 999_99_9999_L;         // Invalid; cannot put underscores prior to an L suffix
+```
```
Comment (SleeplessByte) [inline concepts/numbers/about.md:124]: This doesn't look like JavaScript!

Comment (cydeb33) [reply concepts/numbers/about.md:124]: You're very right indeed, sorry for the mixup I'll fix that

File: concepts/numbers/about.md [hunk 2]
```diff
@@ -55,6 +55,75 @@ There are three types of maximum (and minimum / maximum negative) values for num
...
+
+## NaN or Not-a-Number
+
+The [`NaN`][NaN] is a variable in global scope. There are five different types of operations returning `NaN`:
+
+-  Number cannot be parsed (e.g. parseInt("blabla") or Number(undefined))
+-  Math operation where the result is not a real number (e.g. Math.sqrt(-1))
+-  Operand of an argument is NaN (e.g. 7 ** NaN)
+-  Indeterminate form (e.g. 0 * Infinity, or undefined + undefined)
+-  Any operation that involves a string and is not an addition operation (e.g. "foo" / 3)
```
Comment (junedev) [inline concepts/numbers/about.md:89]: This content is a 1to1 copy from MDN. If you copy content from external sources you need to do so as a quote and mention the source. You can see an example how to do this here:
https://github.com/exercism/javascript/blob/main/concepts/array-analysis/introduction.md?plain=1#L33
https://github.com/exercism/javascript/blob/main/concepts/array-analysis/introduction.md?plain=1#L85

This comment applies to all the content you added. If you copied content, make sure to either rewrite things in your own words or use quotes with sources.

File: concepts/numbers/about.md [hunk 3]
```diff
@@ -55,6 +55,75 @@ There are three types of maximum (and minimum / maximum negative) values for num
...
+- `POSITIVE_INFINITY`, divided by any negative value except `NEGATIVE_INFINITY`, is `NEGATIVE_INFINITY`.
+- Zero multiplied by `POSITIVE_INFINITY` is `NaN`.
+
+[`negative-infinity`][negative-infinity] is the same as `POSITIVE_INFINITY` but with a negative sign :
+- `NEGATIVE_INFINITY`, divided by either `NEGATIVE_INFINITY` or `POSITIVE_INFINITY`, is `NaN`.
+- `NEGATIVE_INFINITY`, divided by any negative value except `NEGATIVE_INFINITY`, is `POSITIVE_INFINITY`.
+- Any positive value, including `POSITIVE_INFINITY`, multiplied by `NEGATIVE_INFINITY` is `NEGATIVE_INFINITY`.
+
+
+## NaN or Not-a-Number
```
Comment (junedev) [inline concepts/numbers/about.md:81]: "or" seems confusing here, maybe "## NaN - Not-a-Number" would be better

File: concepts/numbers/about.md [hunk 4]
```diff
@@ -55,6 +55,75 @@ There are three types of maximum (and minimum / maximum negative) values for num
...
+
+The [`NaN`][NaN] is a variable in global scope. There are five different types of operations returning `NaN`:
```
Comment (junedev) [inline concepts/numbers/about.md:83]: This part is missing a general explanation what NaN actually means/represent. E.g. similar to how it is explained on javascript.info https://javascript.info/types#number:
> `NaN` represents a computational error. It is a result of an incorrect or an undefined mathematical operation

File: concepts/numbers/about.md [hunk 5]
```diff
@@ -55,6 +55,75 @@ There are three types of maximum (and minimum / maximum negative) values for num
 
 Because of how numbers in JavaScript are implemented, **not** every number between `Number.MIN_VALUE` and `Number.MAX_VALUE` can be represented.
 However, _every_ number between `Number.MIN_SAFE_INTEGER - 1` and `Number.MAX_SAFE_INTEGER + 1` **can** be represented.
+## Infinity - Positive and Negative
```
Comment (junedev) [inline concepts/numbers/about.md:58]: Currently your headlines feel like they have different levels. "Built-in Objects" is an h2 header and so is "Infitity" or "NaN" although one a very wide field and one is a very detailed small-ish aspect.

This point is a bit hard to explain but maybe you understand what I mean. I would recommend you create one h2 `Special Built-In "Numbers"` and but the content for infinity and NaN below.

File: concepts/numbers/about.md [hunk 6]
```diff
@@ -55,6 +55,75 @@ There are three types of maximum (and minimum / maximum negative) values for num
...
+```
+Beware that [`positive-infinity`][positive-infinity] behaves differently than the mathematical infinity. So you may encounter some unexpected behaviors like this:
+- `POSITIVE_INFINITY`, divided by either `NEGATIVE_INFINITY` or `POSITIVE_INFINITY`, is `NaN`.
+- `POSITIVE_INFINITY`, divided by any negative value except `NEGATIVE_INFINITY`, is `NEGATIVE_INFINITY`.
+- Zero multiplied by `POSITIVE_INFINITY` is `NaN`.
+
+[`negative-infinity`][negative-infinity] is the same as `POSITIVE_INFINITY` but with a negative sign :
+- `NEGATIVE_INFINITY`, divided by either `NEGATIVE_INFINITY` or `POSITIVE_INFINITY`, is `NaN`.
+- `NEGATIVE_INFINITY`, divided by any negative value except `NEGATIVE_INFINITY`, is `POSITIVE_INFINITY`.
+- Any positive value, including `POSITIVE_INFINITY`, multiplied by `NEGATIVE_INFINITY` is `NEGATIVE_INFINITY`.
```
Comment (junedev) [inline concepts/numbers/about.md:78]: This section (line 75-78) does not add much value compared to the one above. You can replace it with a sentence like "Similar rules apply for negative infinity."

**Review by junedev — CHANGES_REQUESTED** [review]
Thanks for taking a stab here. I left some comments, e.g. on how to better integrate things into the existing content.

In general, please keep in mind that the focus for Exercism is on gaining [fluency](https://exercism.org/docs/using/product/fluency) in a language. This also applies for the concept docs. So while we want to cover the concept exhaustively in about.md, we don't want to go into detail about aspects that are not relevant to write good code in that language. See this section from the [Exercism docs](https://exercism.org/docs/building/tracks/concepts#h-file-about-md) for reference:
> It is **not** the aim of the about.md file to provide a complete set of information on the Concept. As an example, imagine a language that has some older features for which experienced programmers (and maybe even the official docs/specs) recommend they should not be used anymore. Providing details on such features would be out of scope for the about.md file because they are not relevant to gain fluency.

File: concepts/numbers/links.json [hunk 7]
```diff
@@ -14,5 +14,28 @@
...
+  {
+    "url": "https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/POSITIVE_INFINITY",
+    "description": "MDN: Positive infinity"
+  },
+  {
+    "url": "https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/NEGATIVE_INFINITY",
+    "description": "MDN: Negative infinity"
+  },{
+    "url": "https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/NaN",
+    "description": "MDN: Not-A-Number (Built-in object"
```
[the remark this answers is not in the document]
Comment (cydeb33) [reply concepts/numbers/links.json:39]: Oh okay I understand better now, I thought that the links.json was a list retrieving all the links used in the about file

File: concepts/numbers/about.md (hunk 2 above)
Comment (cydeb33) [reply concepts/numbers/about.md:89]: I shall update this accordingly

File: concepts/numbers/about.md [hunk 8]
```diff
@@ -55,6 +55,75 @@ There are three types of maximum (and minimum / maximum negative) values for num
...
+
+The [`NaN`][NaN] is a variable in global scope. There are five different types of operations returning `NaN`:
+
+-  Number cannot be parsed (e.g. parseInt("blabla") or Number(undefined))
+-  Math operation where the result is not a real number (e.g. Math.sqrt(-1))
+-  Operand of an argument is NaN (e.g. 7 ** NaN)
+-  Indeterminate form (e.g. 0 * Infinity, or undefined + undefined)
+-  Any operation that involves a string and is not an addition operation (e.g. "foo" / 3)
+
+## Exponentiation
```
[the remark this answers is not in the document]
Comment (cydeb33) [reply concepts/numbers/about.md:91]: Do you mean by that that it shouldn't be at all on this file?

File: concepts/numbers/about.md [hunk 9]
```diff
@@ -55,6 +55,75 @@ There are three types of maximum (and minimum / maximum negative) values for num
...
+
+// binary
+let nibbles = 0b1011_0101_0101;
+
+// octal
+let val = 0o1234_5670;
+
+// hex
+let message = 0xD0_E0_F0;
+```
```
[the remark this answers is not in the document]
Comment (cydeb33) [reply concepts/numbers/about.md:126]: Makes sense :) !

File: concepts/numbers/about.md (hunk 6 above)
Comment (cydeb33) [reply concepts/numbers/about.md:78]: Ok thanks for the tip

File: concepts/numbers/about.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (junedev) [reply concepts/numbers/about.md:91]: Yes, exactly.

**Comment by junedev** [comment]
@cydeb33 Could you let us know whether you are planning to address the remaining comments so this can be merged? I can see your comments but no commits regarding those yet.

**Comment by cydeb33** [comment]
> @cydeb33 Could you let us know whether you are planning to address the remaining comments so this can be merged? I can see your comments but no commits regarding those yet.

Hello, yes sorry I got really busy with my finals, I do intend to finish up as soon as possible

**Comment by cydeb33** [comment]
@junedev Hello, an unexpected event popped and I won't have sufficient free time to finish this task at the level of quality I intented. My apologies and thank you for your help and patience throughout this experience.

**Comment by junedev** [comment]
@cydeb33 Thanks for letting us know. I will close this PR and leave a note in the issue that your work can be used as a starting point in the future when someone else works on the issue.

**Pull request closed without merging**